src/main/java/net/imagej/ops/deconvolve/RichardsonLucyNonCirculantC.java

```diff
@@ -0,0 +1,338 @@
+			RandomAccessibleInterval.class, RandomAccessibleInterval.class,
+			getFFTInput(), getFFTKernel());
+
+		divide = (BinaryComputerOp) Computers.binary(ops(), DivideHandleZero.class,
+			RandomAccessibleInterval.class, RandomAccessibleInterval.class,
+			RandomAccessibleInterval.class);
+
+	}
+
+	@Override
+	public void performIterations(RandomAccessibleInterval<I> in,
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		createReblurred();
+
+		for (int i = 0; i < getMaxIterations(); i++) {
+
+			if (status != null) {
+				status.showProgress(i, getMaxIterations());
+			}
+
+			// compute correction factor
+			rlCorrection.compute2(in, getRAIExtendedReblurred(),
+				getRAIExtendedReblurred());
+
+			// perform update
+			update.compute1(getRAIExtendedReblurred(), getRAIExtendedEstimate());
+
+			// normalize for non-circulant deconvolution
+			divide.compute2(getRAIExtendedEstimate(), normalization,
+				getRAIExtendedEstimate());
+
+			// accelerate
+			if (getAccelerator() != null) {
+				getAccelerator().mutate(getRAIExtendedEstimate());
+			}
+
+			// create reblurred for the next iteration (so it is available for error
+			// calculation at this iteration)
+			createReblurred();
+
+		}
+	}
+
+	@Override
+	public void preProcess(RandomAccessibleInterval<I> in,
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		Type<O> outType = Util.getTypeFromInterval(out);
+
+		// create image for the estimate, this image is defined over the entire
+		// convolution interval
+		Img<O> estimate = this.getImgFactory().create(getImgConvolutionInterval(),
+			outType.createVariable());
+
+		// set first guess to be a constant = to the average value
+
+		// so first compute the sum...
+		final O sum = ops().stats().<I, O> sum(Views.iterable(in));
+
+		// then the number of pixels
+		final long numPixels = k.dimension(0) * k.dimension(1) * k.dimension(2);
+
+		// then the average value...
+		final double average = sum.getRealDouble() / (numPixels);
+
+		// set first guess as the average value computed above (TODO: make this an
+		// op)
+		for (final O type : estimate) {
+			type.setReal(average);
+		}
+
+		// create image for the reblurred
+		Img<O> reblurred = this.getImgFactory().create(getImgConvolutionInterval(),
+			outType.createVariable());
+
+		setRAIExtendedEstimate(estimate);
+		setRAIExtendedReblurred(reblurred);
+
+		// perform fft of input
+		ops().filter().fft(getFFTInput(), in);
+
+		// perform fft of psfs
+		ops().filter().fft(getFFTKernel(), kernel);
+
+		normalization = getImgFactory().create(estimate, outType.createVariable());
+
+		this.createNormalizationImageSemiNonCirculant(kernel);
+
+	}
+
+	/**
+	 * postProcess TODO: review this function
+	 */
+	@Override
+	protected void postProcess(RandomAccessibleInterval<I> in,
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		// when doing non circulant deconvolution we need to crop and copy back to
+		// the
```

Comment:
In most cases yes, but for the Non-circulant algorithm, the estimate RAI is extended by zero, then the algorithm estimates values outside the border, so you have to use "real" extension, rather than "virtual" extension.